building-and-packaging/material/packaging_python_slides.md

```diff
@@ -0,0 +1,294 @@
+- There are two ways to create these archives:
+    - Using the package builder [build](https://pypa-build.readthedocs.io/en/stable/index.html) which was introduced in [PEP 517](https://www.python.org/dev/peps/pep-0517/)
+    - Using the packaging standard [wheel](https://wheel.readthedocs.io/en/stable/) which was introduced in [PEP 427](https://www.python.org/dev/peps/pep-0427/)
+- A source distribution archive file (`package-name-<tags>.tar.gz`) can be generated by running the command `python3 setup.py sdist` in the `package_name/` repository
+- A wheel archive file (`package-name-<tags>.whl`) can be generated by running the command `python3 setup.py bdist_wheel` in the `package_name/` repository
+- In both cases the files will be generated in a folder `package_name/dist/`
+
+---
+
+## Uploading the distribution archives 1/2
+
+- [Twine](https://twine.readthedocs.io/en/latest/) is a common tool to upload our package to PyPI
+- Why Twine?
+    - Secure authentication of the user to PyPI over HTTPS using a verified connection
+    - Its predecessor `python setup.py upload` required careful configuration
+    - Encourages users to create distribution files to promote testing before releasing
+- [TestPyPI](https://test.pypi.org/) is a testing instance of PyPI which does not affect the real index
+- Uploading the distribution to TestPyPI before PyPI is a standard pre-deployment step
+
+---
+
+## Uploading the distribution archives 2/2
+
+- Before using TestPyPI (or PyPI) you need an account and a PyPI API Token
+- Account creation and API Token generation is straightforward through the [registration page](https://test.pypi.org/account/register/)
+- Uploading to TestPyPI can be done using the command: `python3 -m twine upload --repository testpypi dist/*`
+- The uploading process looks something like:
+
+```bash
+Uploading distributions to https://test.pypi.org/legacy/
+Enter your username: [your username]
+Enter your password: [your password]
+Uploading package_name-0.0.1-py3-none-any.whl
+100%|█████████████████████| 4.65k/4.65k [00:01<00:00, 2.88kB/s]
+Uploading package_name-0.0.1.tar.gz
+100%|█████████████████████| 4.25k/4.25k [00:01<00:00, 3.05kB/s]
+```
+
+---
+
+## Installing the uploaded package
+
+- The last step in the packaging pipeline is a sanity check done by installing the package that has been uploaded
+- The installation can be done by:
+
+```bash
+pip install --user --index-url https://test.pypi.org/simple/ <package-name>
+```
+
+---
+
+## Conda
+
+- [Conda](https://docs.conda.io/en/latest/) is a package and environment management system which supports multiple languages
+- Conda provides a fast option to setup an isolated environment on your local system
+- Conda is configured to work with [Anaconda installers and packages](https://repo.anaconda.com/)
+- Conda is often the preferred way to run Python packages on Windows and MacOS. See example in [FEniCS on Anaconda section](https://fenicsproject.org/download/)
```

Comment:
Is it not popular on Linux? I have only used it on Linux so far. :smile:

Reply:
Hence the *often*, meaning that excluding Linux it is also preferred in ....